### PR TITLE
Styling of MAT and school dashboards

### PR DESCRIPTION
**Pull request by david-mears-dfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-1184

## Changes in this PR:

- Fix capitalization and pluralization of MAT dashboard headings such as '1 Published jobs' -> '1 published job'

Styling changes:
- Correctly position green 'Create' button for school users (in line with school name)
- Allow overflow scroll for vacancies table, using MoJ component
- Use thirds instead of quarters for columns
- Make vacancies table visible on mobile

## TVS Design System templates for reference for non-trust dashboard

https://www.figma.com/file/8BS7fuTGkUxx3ReqKdkufm/TVS-Design-System?node-id=83%3A796

## Prototype

https://dfe-prototype.herokuapp.com/mat3/dashboard

## Screenshots of UI changes

| Before | After |
------- | ------
| MAT dashboard with less-wide desktop | Allow overflow scroll for vacancies table, using MoJ component; Use thirds instead of quarters for columns |
| Screenshot 2020-08-26 at 23 25 28 | Screenshot 2020-08-27 at 11 23 58 |
| School dashboard with less-wide desktop | Allow overflow scroll for vacancies table, using MoJ component; Correctly position green 'Create' button for school users (in line with school name) |
| Screenshot 2020-08-26 at 23 30 10 | Screenshot 2020-08-26 at 23 27 07 |
| Mobile view: table is not visible | Mobile view: table is visible |
| Screenshot 2020-08-26 at 23 29 27 | Screenshot 2020-08-26 at 23 27 55 |

## Next steps:

- [x] UX Sign-off

### CODE REVIEW

**Review by alexbowen — APPROVED** [review]
i will have to make an associated change quarter -> third in my PR but very easy to do and i suggest this is merged first

File: app/views/hiring_staff/organisations/_vacancies_awaiting_feedback.html.haml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-%section.govuk-tabs__panel
-  %h2.govuk-heading-m= t("jobs.awaiting_feedback_jobs_with_count", count: vacancies.count)
+%section.govuk-tabs__panel.desktop-container-width
+  %h2.govuk-heading-m= t("jobs.awaiting_feedback_jobs_with_count", count: vacancies.count, jobs: vacancies.count == 1 ? t("jobs.job") : t("jobs.job").pluralize)
```
Comment (cpjmcquillan) [inline app/views/hiring_staff/organisations/_vacancies_awaiting_feedback.html.haml]: Can we use `zero:..., one:..., other:...` in the translation file and lose the template logic?

File: app/components/hiring_staff/inline_components/vacancies_count_heading_component.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class HiringStaff::InlineComponents::VacanciesCountHeadingComponent < ViewComponent::Base
+  def initialize(vacancies: vacancies, selected_type: selected_type)
+    @vacancies = vacancies
+    @selected_type = selected_type
+  end
+
+  def call
+    t("jobs.#{@selected_type}_jobs_with_count",
```
Comment (cpjmcquillan) [inline app/components/hiring_staff/inline_components/vacancies_count_heading_component.rb:8]: I think my earlier comment is still applicable, but it's slightly less pertinent now we're out of the template

File: app/components/hiring_staff/inline_components/vacancies_count_heading_component.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+      count: @vacancies.count,
+      jobs: @vacancies.count == 1 ? t('jobs.job') : t('jobs.job').pluralize)
```
Comment (cpjmcquillan) [inline app/components/hiring_staff/inline_components/vacancies_count_heading_component.rb:10]: In fact if you don't use the translation file to do it, this component probably needs a test

**Review by cpjmcquillan — APPROVED** [review]
One minor todo with the translation but nice work 👍 . Also, in the screenshot the MAT title seems to have wrapped despite there seeming to be sufficient room for it not to?

File: app/components/hiring_staff/inline_components/vacancies_count_heading_component.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  def call
+    case @vacancies.count
+    when 0
+      t("jobs.#{@selected_type}_jobs_with_count.zero")
+    when 1
+      t("jobs.#{@selected_type}_jobs_with_count.one")
+    else
+      t("jobs.#{@selected_type}_jobs_with_count", count: @vacancies.count)
```
Comment (cpjmcquillan) [inline app/components/hiring_staff/inline_components/vacancies_count_heading_component.rb:14]: This alone is sufficient, the translation uses the count to decide which one to use itself

File: config/locales/en.yml [hunk 5]
```diff
@@ -469,20 +469,35 @@ en:
     share_on_twitter: Share to Twitter
     share_this_job: Share this job
     published_jobs: Published jobs
-    published_jobs_with_count: '%{count} Published jobs'
+    published_jobs_with_count:
+      zero: 'No published jobs'
+      one: '1 published job'
+      other: '%{count} published %jobs'
```
Comment (cpjmcquillan) [inline config/locales/en.yml:475]: `other: %{count} published jobs`

**Comment by david-mears-dfe** [comment]
Fixed translation and wrapping.

**Review by cpjmcquillan — APPROVED** [review]

File: app/frontend/styles/application.scss [hunk 6]
```diff
@@ -34,6 +34,7 @@ $govuk-image-url-function: frontend-image-url;
 @import 'components/links';
 @import 'components/location_finder';
 @import 'components/modal_dialogue';
+@import 'components/moj_scrollable_pane';
```
Comment (alexbowen) [inline app/frontend/styles/application.scss:37]: isnt this already imported on line 13 of this file?

Comment (alexbowen) [reply app/frontend/styles/application.scss:37]: ignore that :)

**Review by alexbowen — APPROVED** [review]

File: app/frontend/styles/application.scss (hunk 6 above)
Comment (david-mears-dfe) [reply app/frontend/styles/application.scss:37]: That's the original unedited version

**Pull request merged**